tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -551,6 +627,37 @@ where
     }
 }
 
+struct Timings {
+    accumulated: u64,
+    entered: Option<Instant>,
+}
+
+impl Timings {
+    fn new() -> Self {
+        Self {
+            accumulated: 0,
+            entered: Some(Instant::now()),
+        }
+    }
+    fn get_accumulated(&self) -> String {
+        if self.accumulated >= 10_000_000_000 {
+            format!("{}s", self.accumulated)
+        } else if self.accumulated >= 1_000_000_000 {
+            format!("{:.1}s", self.accumulated as f64 / 1_000_000_000.0)
+        } else if self.accumulated >= 10_000_000 {
+            format!("{}ms", self.accumulated / 1_000_000)
+        } else if self.accumulated >= 1_000_000 {
+            format!("{:.1}ms", self.accumulated as f64 / 1_000_000.0)
+        } else if self.accumulated >= 10_000 {
+            format!("{}µs", self.accumulated / 1000)
+        } else if self.accumulated >= 1000 {
+            format!("{:.1}µs", self.accumulated as f64 / 1000.0)
+        } else {
+            format!("{}ns", self.accumulated)
+        }
+    }
```

Comment:
I think we could do all this without allocating if we changed this to return `impl fmt::Debug` and used `format_args!` rather than `format!`, or if we put it in a `fmt::Debug` impl.

Reply:
Yes, of course; I haven’t much optimised for allocations in this hack :-)